### PR TITLE
update gemma recipe

### PR DESCRIPTION
**Pull request by wenhuach21**

### CODE REVIEW

**Review by yintong-lu — APPROVED** [review]

File: auto_round/export/export_to_autoround/autoround_quantizer.py [hunk 1]
```diff
@@ -191,22 +194,22 @@ class AutoRoundConfig(QuantizationConfigMixin):
...
-        n_samples=512,
-        seqlen=2048,
-        **kwargs,
+            self,
+            bits: int,
+            tokenizer: Any = None,
+            dataset: str = None,
+            group_size: int = 128,
+            sym: bool = False,
+            backend="gptq:exllamav2",
```
Comment (WeiweiZhang1) [inline auto_round/export/export_to_autoround/autoround_quantizer.py:203]: Mind the indentation

File: README.md [hunk 2]
```diff
@@ -142,7 +142,7 @@ print(tokenizer.decode(model.generate(**inputs, max_new_tokens=50)[0]))
 
 ```python
 from transformers import AutoModelForCausalLM, AutoTokenizer
-
+##from auto_round.export import AutoHfQuantizer ## uncomment it for models with quantized lm-head
```
Comment (WeiweiZhang1) [inline README.md:145]: # Please uncomment this line for models with quantized lm-head

File: docs/gemma-7b-acc.md [hunk 3]
```diff
@@ -1,22 +1,23 @@
-Install [lm-eval-harness](https://github.com/EleutherAI/lm-evaluation-harness.git) from source,  and the  git id 96d185fa6232a5ab685ba7c43e45d1dbb3bb906d, Install the latest [AutoGPTQ](https://github.com/AutoGPTQ/AutoGPTQ) from source first
```
Comment (WeiweiZhang1) [inline docs/gemma-7b-acc.md:1]: Install [lm-eval-harness](https://github.com/EleutherAI/lm-evaluation-harness.git) from source, lm-eval 0.4.2 is used

File: examples/language-modeling/main.py [hunk 4]
```diff
@@ -121,6 +121,9 @@
     parser.add_argument("--quant_lm_head", action='store_true',
                         help="quant_lm_head")
 
+    parser.add_argument("--model_dtype", default=None, type=str,
+                        help="force to convert the dtype, some backends supports fp16 dtype better")
```
Comment (WeiweiZhang1) [inline examples/language-modeling/main.py:125]: Forcibly specify the dtype to be used for quantization.

File: examples/language-modeling/main.py [hunk 5]
```diff
@@ -212,6 +215,12 @@ def get_library_version(library_name):
         model.seqlen = seqlen
     seqlen = args.seqlen
 
+    if args.model_dtype != None:
+        if args.model_dtype == "float16" or args.model_dtype == "fp16":
+            model = model.to(torch.float16)
+        if args.model_dtype == "bfloat16" or args.model_dtype == "bfp16":
+            model = model.to(torch.bfloat16)
```
Comment (WeiweiZhang1) [inline examples/language-modeling/main.py:222]: how about fp32

**Review by WeiweiZhang1 — APPROVED** [review]

**Pull request merged**